pkg/api/vlabs/validate.go

```diff
@@ -271,6 +271,17 @@ func (a *Properties) validateOrchestratorProfile(isUpdate bool) error {
 							minVersion.String(), o.OrchestratorVersion)
 					}
 				}
+				// If the user assigned identity is enabled, then make sure the version is higher than v1.12.0-alpha.1
+				if o.KubernetesConfig.UseManagedIdentity && o.KubernetesConfig.UserAssignedID != "" {
```

Comment:
Could you add unit tests for these new validations?

Reply:
This validation is now here: https://github.com/Azure/acs-engine/blob/master/pkg/api/vlabs/validate.go#L672-L674.